src/main/java/io/excaliburfrc/robot/subsystems/Climber.java

```diff
@@ -94,6 +106,51 @@ public void closeAngler() {
     anglerPiston.set(DoubleSolenoid.Value.kReverse);
   }
 
+  private void achieveState(TrapezoidProfile.State setpoint) {
+    ElevatorFeedforward ff =
+        anglerPiston.get() == DoubleSolenoid.Value.kForward ? diagonalFF : upFF;
+    leftController.setReference(
+        setpoint.velocity,
+        ControlType.kVelocity,
+        0,
+        ff.calculate(setpoint.velocity),
+        ArbFFUnits.kVoltage);
+    rightController.setReference(
+        setpoint.velocity,
+        ControlType.kVelocity,
+        0,
+        ff.calculate(setpoint.velocity),
+        ArbFFUnits.kVoltage);
+  }
+
+  private Command reachBarCommand(Consumer<TrapezoidProfile.State> toRun) {
+    return new TrapezoidProfileCommand(elevatorProfile, toRun, this);
+  }
+
+  public Command openToFirstCommand() {
+    return reachBarCommand(this::achieveState);
+  }
+
+  public Command openToSecondCommand() {
+    return reachBarCommand(
+        setpoint -> {
+          if (setpoint.position >= HEIGHT_TO_OPEN_PISTON
+              && anglerPiston.get() != DoubleSolenoid.Value.kForward) {
+            anglerPiston.set(DoubleSolenoid.Value.kForward);
+          }
+        });
+  }
+
+  public Command raiseRobotCommand() {
+    return new ParallelCommandGroup(
+        new RunCommand(() -> leftMotor.set(-1), this)
+            .withInterrupt(() -> leftEncoder.getPosition() <= SAFETY_DISTANCE)
+            .andThen(() -> leftMotor.set(0)),
+        new RunCommand(() -> rightMotor.set(-1))
+            .withInterrupt(() -> rightEncoder.getPosition() <= SAFETY_DISTANCE)
+            .andThen(() -> rightMotor.set(0)));
```

Comment:
can you split this to a function that returns a per-side command method? and this can use `FunctionalCommand` instead